packages/web/src/features/languages/ImportLanguageGlossesView.tsx

```diff
@@ -32,29 +25,89 @@ export async function importLanguageGlossesLoader({
     params.code ?? 'unknown'
   );
   const importLanguages = await apiClient.import.getLanguages();
-  return { language, importLanguages };
+  let currentJob;
+  try {
+    currentJob = await apiClient.languages.getImportStatus(
+      params.code ?? 'unknown'
+    );
+  } catch (error) {
+    if (!(error instanceof ApiClientError) || error.status !== 404) {
+      throw error;
+    }
+  }
+  return { language, importLanguages, currentJob };
 }
 
 interface FormData {
   import: string;
 }
 
 export default function ImportLanguageGlossesView() {
-  const { language, importLanguages } = useLoaderData() as {
-    language: GetLanguageResponseBody;
-    importLanguages: GetLanguageImportOptionsResponseBody;
-  };
+  const { language, importLanguages, currentJob } = useLoaderData() as Awaited<
+    ReturnType<typeof importLanguageGlossesLoader>
+  >;
   const flash = useFlash();
   const { t } = useTranslation();
   const confirmationDialog = useRef<ConfirmationDialogRef>(null);
 
   const formContext = useForm<FormData>();
+
+  const [importStatus, setImportStatus] = useState<
+    'idle' | 'running' | 'complete' | 'error'
+  >(() => {
+    if (currentJob) {
+      if (currentJob.endDate) {
+        if (
+          Date.now() - new Date(currentJob.endDate).valueOf() <
+          10 * 60 * 1000 // 10 minutes
+        ) {
+          return 'complete';
+        } else {
+          return 'idle';
+        }
+      } else {
+        return 'running';
+      }
+    } else {
+      return 'idle';
+    }
+  });
+  const pollTimeout = useRef<NodeJS.Timer>();
+  useEffect(() => {
+    if (importStatus === 'running') {
+      const checkImportStatus = async () => {
+        const job = await apiClient.languages.getImportStatus(
+          language.data.code
+        );
```

Comment:
When the job is running, the frontend polls the backend for the job status on a five second interval.